blockchain/chaingen/generator.go

```diff
@@ -962,16 +930,15 @@ func (g *Generator) CalcNextRequiredStakeDifficulty() int64 {
 	targetPoolSize := ticketsPerBlock * int64(g.params.TicketPoolSize)
 	numWindows := g.params.StakeDiffWindows
 	weightAlpha := g.params.StakeDiffAlpha
-	block := g.tip
 	for i := int64(0); i < numWindows; i++ {
 		// Get the pool size for the block at the start of the window.
 		// Use zero if there are not yet enough blocks left to cover the
 		// window.
 		prevRetargetHeight := nextHeight - uint32(windowSize*(i+1))
 		windowPoolSize := int64(0)
-		block = g.ancestorBlock(block, prevRetargetHeight, nil)
-		if block != nil {
-			windowPoolSize = int64(block.Header.PoolSize)
+		ancestorBlock := g.ancestorBlock(prevBlock, prevRetargetHeight, nil)
```

Comment:
This isn't correct.  It's not going back to the correct ancestor each iteration, rather only going back to the same one.  This should be left as it was, and ring back the `block := g.tip` above, except make it `block := prevBlock`.